drivers/ethernet/eth_stm32_hal.c

```diff
@@ -54,42 +54,12 @@ LOG_MODULE_REGISTER(LOG_MODULE_NAME);
 #error DTCM for DMA buffer is activated but zephyr,dtcm is not present in dts
 #endif
 
-#define PHY_ADDR	CONFIG_ETH_STM32_HAL_PHY_ADDRESS
+static const struct device *eth_stm32_phy_dev = DEVICE_DT_GET(DT_INST_PHANDLE(0, phy_handle));
 
-#if defined(CONFIG_ETH_STM32_HAL_API_V2)
-#define PHY_SCSR               ((uint16_t)0x001FU) /*!< PHY Special Control/Status */
-#define PHY_SCSR_AUTONEGO_DONE ((uint16_t)0x1000U) /*!< Auto-Negotiation Done Status */
-#define PHY_HCDSPEEDMASK       ((uint16_t)0x001CU) /*!< High Capability Speed Mask */
-#define PHY_10BT_HD            ((uint16_t)0x0004U) /*!< 10Base-T half-duplex */
-#define PHY_10BT_FD            ((uint16_t)0x0014U) /*!< 10Base-T full-duplex */
-#define PHY_100BTX_HD          ((uint16_t)0x0008U) /*!< 100Base-TX half-duplex */
-#define PHY_100BTX_FD          ((uint16_t)0x0018U) /*!< 100Base-TX full-duplex */
-#define PHY_AUTONEGO_ENABLE    ((uint16_t)0x1000U) /*!< Auto-negotiation enable bit */
-#define PHY_TIMEOUT            (5000U)             /*!< PHY operation timeout in msec */
-
-#define PHY_STATUS_LINK_DOWN           ((int32_t)1) /*!< Link down status */
-#define PHY_STATUS_100MBITS_FULLDUPLEX ((int32_t)2) /*!< 100 Mbps full-duplex status */
-#define PHY_STATUS_100MBITS_HALFDUPLEX ((int32_t)3) /*!< 100 Mbps half-duplex status */
-#define PHY_STATUS_10MBITS_FULLDUPLEX  ((int32_t)4) /*!< 10 Mbps full-duplex status */
-#define PHY_STATUS_10MBITS_HALFDUPLEX  ((int32_t)5) /*!< 10 Mbps half-duplex status */
-#define PHY_STATUS_AUTONEGO_NOTDONE    ((int32_t)6) /*!< Auto-negotiation not done */
-#endif
-
-#if DT_HAS_COMPAT_STATUS_OKAY(st_stm32_mdio)
-
-#define DEVICE_PHY_BY_NAME(n) \
-	    DEVICE_DT_GET(DT_CHILD(DT_INST_CHILD(n, mdio), _CONCAT(ethernet_phy_, PHY_ADDR)))
-
-static const struct device *eth_stm32_phy_dev = DEVICE_PHY_BY_NAME(0);
-
-#endif
+#define ETH_STM32_AUTO_NEGOTIATION_ENABLE                                                          \
+	UTIL_NOT(DT_NODE_HAS_PROP(DT_INST_PHANDLE(0, phy_handle), fixed_link))
```

Comment:
[Per `DT_NODE_HAS_PROP`'s documentation](https://github.com/zephyrproject-rtos/zephyr/blob/ca0f5f6924cdc7046ce8dbb6d32e39125b5ded7f/include/zephyr/devicetree.h#L3771-L3785):
```
To get a boolean property's truth value, use DT_PROP(node_id, prop) instead.
```
```suggestion
	UTIL_NOT(DT_PROP(DT_INST_PHANDLE(0, phy_handle), fixed_link))
```

Reply:
from ethernet-phy.yaml:
```
  fixed-link:
    type: string
    description: This link is fixed and does not require PHY configuration
    enum:
      - "10BASE-T Half-Duplex"
      - "10BASE-T Full-Duplex"
      - "100BASE-T Half-Duplex"
      - "100BASE-T Full-Duplex"
      - "1000BASE-T Half-Duplex"
      - "1000BASE-T Full-Duplex"
```
it is not a bool

Reply:
Oh. Nevermind then 🙇